### ISSUE TITLE
Unhandled FormatException for invalid querystring

### ISSUE THREAD

**Issue by jonasfj** [dart-lang/pub-dev#3372]
From logs:
```
A 2020-02-25T02:41:25Z ERROR - 2020-02-25 02:41:25.497
A 2020-02-25T02:41:25Z GET /publishers/dart.dev/packages?q=xXXXXX&legacy=1&sort=XXX.com%FFx@11.111.11.11
A 2020-02-25T02:41:25Z Error thrown by handler.
A 2020-02-25T02:41:25Z FormatException: Bad UTF-8 encoding 0xff (at offset 20)
A 2020-02-25T02:41:25Z dart:core                                                _Uri.queryParameters
A 2020-02-25T02:41:25Z package:pub_dev/shared/handler_helpers.dart 87:30        _requestContextWrapper.<fn>
```
I think we need to have our logging wrapper higher up, maybe we just reorder our middlewares.

**Comment by isoos** [comment]
Yeah, looks like these are called outside of the log wrapper:
https://github.com/dart-lang/pub-dev/blob/master/app/lib/shared/handler_helpers.dart#L74-L79

Coincidentally, #3373 has a suspected method that is also outside of that logging.

**Comment by isoos** [comment]
Fixed by #3524.
